mautic/mautic
# A Contact Point Log report can be generated with a segment filter

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [&check;]
| New feature/enhancement? (use the a.x branch)      | [&cross;]
| Deprecations?                          | [&cross;]
| BC breaks? (use the c.x branch)        | [&cross;]
| Automated tests included?              | [&cross;] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:


Filters and graphs included simultaneously inthe Contact Point Log report lead to an error.
```
"message": "An exception occurred while executing 'SELECT DATE_FORMAT(lp.date_added, '%Y-%m-%d') AS date, COUNT() AS count FROM lead_points_change_log lp LEFT JOIN leads l ON l.id = lp.lead_id WHERE (lp.date_added IS NULL OR (lp.date_added BETWEEN ? AND ?)) AND ((l.firstname IS NOT NULL) AND (l.firstname <> '') AND (l.company NOT LIKE ?) AND (l.company NOT LIKE ?) AND (s.leadlist_id = ?)) GROUP BY DATE_FORMAT(lp.date_added, '%Y-%m-%d') ORDER BY DATE_FORMAT(lp.date_added, '%Y-%m-%d') ASC LIMIT 32' with params ["2023-05-25 22:00:00", "2023-06-26 12:52:54", "aivie", "idea2", "107"]:\n\nSQLSTATE[42S22]: Column not found: 1054 Unknown column 's.leadlist_id' in 'where clause'",
"class": "Doctrine\DBAL\Exception\InvalidFieldNameException",
"previous": {
"message": "SQLSTATE[42S22]: Column not found: 1054 Unknown column 's.leadlist_id' in 'where clause'",
```
The cause of it is removing all join clauses in the code, but remaining all the related where clauses. 
#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new Contact Point Log report.
3. Add any columns, a segment filter and any graphs/tables
![image](https://github.com/mautic/mautic/assets/96085911/656dd4af-8560-4866-a1a8-f8ef3926b0a6)
![image](https://github.com/mautic/mautic/assets/96085911/00b9a79f-3228-4bc3-a10d-49a8227f7f23)
5. Save and close the report.
6. There is an error. This PR fixes it.
